cmd/jaeger/internal/integration/e2e_integration.go

```diff
@@ -152,6 +152,28 @@ func (s *E2EStorageIntegration) e2eInitialize(t *testing.T, storage string) {
 	s.SpanReader, err = createSpanReader(logger, ports.QueryGRPC)
 	require.NoError(t, err)
 
+	req, err := http.NewRequestWithContext(context.Background(), http.MethodGet, "http://localhost:8888/metrics", nil)
```

Comment:
a) why would you scrape metrics _before_ the test starts?
b) please move this code to a helper function
c) you can probably schedule this function to run in `t.Cleanup()` so that it runs just before the binary is stopped

Reply:
Got it